### PR TITLE
Adding a pre-test check that looks for java files in the test/scala d…

### PR DESCRIPTION
**Pull request by yfarjoun**
…irectory and for scala files anywhere else (and fails if it finds either of these)

### Description

Please explain the changes you made here.
Explain the **motivation** for making this change. What existing problem does the pull request solve?

### Checklist

- [ ] Code compiles correctly
- [ ] New tests covering changes and new functionality
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -81,6 +81,23 @@ tasks.withType(Test) { task ->
...
+    jvmArgs += '-Dsamjdk.sra_libraries_download=true'
+
+    tags {
+        exclude "slow"
+        exclude "broken"
+    }
+}
+
+
+task FindScalaAndJavaTypes(type: Exec) {
```
Comment (lbergelson) [inline build.gradle]: I think we usually start task names with lowercase

File: build.gradle [hunk 2]
```diff
@@ -81,6 +81,23 @@ tasks.withType(Test) { task ->
...
+    description = "Check for Scala outside the scala test dir and java files there."
```
Comment (droazen) [inline build.gradle]: "and java files there" -> "and java files in the scala test dir"  (description is a little confusing as-is)

**Comment by droazen** [comment]
:+1: merge after addressing comments

**Comment by yfarjoun** [comment]
verbal 👍  at review party.

**Pull request merged**